### ISSUE TITLE
Make it possible to skip restic backup/restore

### ISSUE THREAD

**Issue by fersingb** [heptio/ark#957]
**Describe the solution you'd like**
I have set up ark to use restic and it works well. But sometimes I'd like to not backup all the data, only the cluster state. In this case it would be nice to have a flag to disable restic. 

The same applies for the restore phase. If I have a backup that has a corresponding restic snaphot, I'd like to be able to ignore that snapshot and restore only the cluster state.

Thank you

**Environment:**

- Ark version (use `ark version`): v0.9.8
- Kubernetes version (use `kubectl version`): v1.12.1
- Kubernetes installer & version: 
- Cloud provider or hardware configuration: Openstack
- OS (e.g. from `/etc/os-release`): CoreOS stable (1800.6.0)

**Comment by Subv** [comment]
I think using `snapshotVolumes: false` in the Schedule definition solves this. When the Backup runs it won't try to snapshot the volumes, just the cluster state.

https://velero.io/docs/v1.5/api-types/schedule/#docs

**Comment by joshkwedar** [comment]
@Subv I tested by specifying `--snapshot-volumes=false` during a backup create, yet the volume annotated for velero backup is still backed up via restic.  I saw the following in the comments of doc you've linked.  This option doesn't appear to work for restic.

** # Whether or not to snapshot volumes. This only applies to PersistentVolumes for Azure, GCE, and
    # AWS. Valid values are true, false, and null/unset. If unset, Velero performs snapshots as long as
    # a persistent volume provider is configured for Velero.**

**Comment by sseago** [comment]
If you add persistentvolumes to ExcludeResources, would that handle this?

**Comment by Lyndon-Li** [comment]
The current requirement might be like this??
Users want to back up heir PV object without backing up the data inside
Users want their PVs to be restored back but don't want to fill data into the PV

Does this make sense?

**Comment by jerry-jibu** [comment]
+1， for example, user may have two types of PVC volumes. One is for logging and the other is for DB. 
For backup, only DB related volumes are required for backup while only volumes/PVC resources for logging should be preserved during backup without copy corresponding data.  It's ideally that we could have a more flexible/convenient way to include/exclude volumes during backup/restore compared the method to annotate volumes on pods one by one.

**Comment by jerry-jibu** [comment]
Another use case is that for HA application with data replication, it will be more efficient to only back up one data copy. 
For example,  we could do only 1 copy of data backup on  either master or slave mysql instance.

**Comment by ywk253100** [comment]
@jerry-jibu Could you try the solution provided by https://github.com/vmware-tanzu/velero/issues/957#issuecomment-1156490764?

**Comment by reasonerjt** [comment]
Moving this out of v1.10 as we don't have the bandwidth per discussion with jibu guys offline.

**Comment by blackpiglet** [comment]
> +1， for example, user may have two types of PVC volumes. One is for logging and the other is for DB. For backup, only DB related volumes are required for backup while only volumes/PVC resources for logging should be preserved during backup without copy corresponding data. It's ideally that we could have a more flexible/convenient way to include/exclude volumes during backup/restore compared the method to annotate volumes on pods one by one.

@jerry-jibu 
For your case, how to distinguish two types of PVC volumes?

**Comment by blackpiglet** [comment]
The original issue should be resolved by Restic opt-in, opt-out way and --exclude-resources parameter.
There are two use cases may need consideration:
* Certain types of PVC should be skipped.
* In some HA cases, different PVs contains same data. Backup only one is enough.
* Some PVs can be mounted multiple times. Should backup only once.

**Comment by blackpiglet** [comment]
> Another use case is that for HA application with data replication, it will be more efficient to only back up one data copy. For example, we could do only 1 copy of data backup on either master or slave mysql instance.

Could you give a use case for this scenario? I can only think of backuping one of the STS's volume for example.
